# Description: use GtkHeaderBar as decoration only under gnome-shell
# Reported as a wishlist on https://bugzilla.gnome.org/show_bug.cgi?id=736420
# The solution should probably be different from the getenv one though
diff -Nur evince-3.13.91.orig/shell/ev-window.c evince-3.13.91/shell/ev-window.c
--- evince-3.13.91.orig/shell/ev-window.c	2014-09-10 15:03:35.645739965 +0200
+++ evince-3.13.91/shell/ev-window.c	2014-09-10 15:07:25.469747618 +0200
@@ -6907,8 +6907,15 @@
 
 	ev_window->priv->toolbar = ev_toolbar_new (ev_window);
 	gtk_widget_set_no_show_all (ev_window->priv->toolbar, TRUE);
-	gtk_header_bar_set_show_close_button (GTK_HEADER_BAR (ev_window->priv->toolbar), TRUE);
-	gtk_window_set_titlebar (GTK_WINDOW (ev_window), ev_window->priv->toolbar);
+	if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME")) {
+		gtk_header_bar_set_show_close_button (GTK_HEADER_BAR (ev_window->priv->toolbar), TRUE);
+		gtk_window_set_titlebar (GTK_WINDOW (ev_window), ev_window->priv->toolbar);
+	}
+	else {
+		gtk_box_pack_start (GTK_BOX (ev_window->priv->main_box),
+					ev_window->priv->toolbar,
+					FALSE, TRUE, 0);
+	}
 	gtk_widget_show (ev_window->priv->toolbar);
 
 	/* Window title */
diff -Nur evince-3.13.91.orig/shell/ev-window-title.c evince-3.13.91/shell/ev-window-title.c
--- evince-3.13.91.orig/shell/ev-window-title.c	2014-09-10 15:03:35.573739962 +0200
+++ evince-3.13.91/shell/ev-window-title.c	2014-09-10 16:27:44.005908089 +0200
@@ -142,7 +142,8 @@
 	switch (window_title->type) {
 	case EV_WINDOW_TITLE_DOCUMENT:
 		gtk_window_set_title (window, title);
-		if (title_header && subtitle) {
+		if (title_header && subtitle && 
+		    !g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME")) {
 			gtk_header_bar_set_title (toolbar, title_header);
 			gtk_header_bar_set_subtitle (toolbar, subtitle);
 		}
@@ -154,8 +155,10 @@
 		gtk_window_set_title (window, password_title);
 		g_free (password_title);
 
+		if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "GNOME")) {
                 gtk_header_bar_set_title (toolbar, _("Password Required"));
                 gtk_header_bar_set_subtitle (toolbar, title);
+        	}
         }
 		break;
         case EV_WINDOW_TITLE_RECENT:

